src/map/options.ts

```diff
@@ -323,12 +323,20 @@ export class MapOptions extends OptionsGroup {
                 );
 
                 // set width first, since setting position can influence it
-                modalDialog.style.width = `${modalDialog.offsetWidth}px`;
+                // scale width of the larger modal-lg class
+                modalDialog.style.width = `${modalDialog.offsetWidth / 1.5}px`;
+                // minimum width so that text in rows remains on a single line
+                modalDialog.style.minWidth = `400px`;
                 // unset margins when using position: fixed
                 modalDialog.style.margin = '0';
                 modalDialog.style.position = 'fixed';
                 modalDialog.style.top = `${top}px`;
-                modalDialog.style.left = `${left}px`;
+
+                if (window.innerWidth < 1400) {
+                    modalDialog.style.right = `10px`;
+                } else {
+                    modalDialog.style.left = `${left}px`;
+                }
```

Comment:
The code in the library should not touch what the user gave in left/top, since the library does not know the overall layout of the different panels. 

Instead this change should live in the app, where we set this layout.